libraries/standard/mqtt/integration-test/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+include("../mqttFilePaths.cmake")
+project ("mqtt unit test")
+cmake_minimum_required (VERSION 3.13)
+
+# ====================  Define your project name (edit) ========================
+set(project_name "mqtt_system")
+
+# ================= Create the library under test here (edit) ==================
+
+# list the files you would like to test here
+list(APPEND real_source_files
+            ${MQTT_SOURCES}
+        )
+# list the directories the module under test includes
+list(APPEND real_include_directories
+            .
+            ${MQTT_INCLUDE_PUBLIC_DIRS}
+            ${MQTT_INCLUDE_PRIVATE_DIRS}
+            ${LOGGING_INCLUDE_DIRS}
+        )
+
+# =====================  Create UnitTest Code here (edit)  =====================
+
+# list the directories your test needs to include
+list(APPEND test_include_directories
+            .
+            ${MQTT_INCLUDE_PUBLIC_DIRS}
+            ${MQTT_INCLUDE_PRIVATE_DIRS}
+            ${LOGGING_INCLUDE_DIRS}
+            # Header path for the TLS server auth utility.
+            ${ROOT_DIR}/demos/transport
+        )
+
+# =============================  (end edit)  ===================================
+set(real_name "${project_name}_real")
+
+create_real_library(${real_name}
+                    "${real_source_files}"
+                    "${real_include_directories}"
+                    # Empty mock name as create_real_library needs the 4th argument.
+                    ""
+        )
+
+list(APPEND utest_link_list
+            lib${real_name}.a
+        )
+
+list(APPEND utest_dep_list
+            ${real_name}
+            tls_utils
+        )
+
+set(utest_name "${project_name}_utest")
+set(utest_source "${project_name}_utest.c")
+create_test(${utest_name}
+            ${utest_source}
+            "${utest_link_list}"
+            "${utest_dep_list}"
+            "${test_include_directories}"
+        )
```

Comment:
hmm this implies that integration tests will affect coverage results for unit tests

Might need to create a separate target for this one.